src/main/java/pl/edu/agh/iet/katabank/bankproduct/interestpolicy/DailyInterestPolicyWithInsurance.java

```diff
@@ -0,0 +1,43 @@
+package pl.edu.agh.iet.katabank.bankproduct.interestpolicy;
+
+
+import java.math.BigDecimal;
+import java.time.Duration;
+import java.time.LocalDate;
+
+public class DailyInterestPolicyWithInsurance implements InterestPolicy {
+
+    private BigDecimal yearlyInterestRatePercent;
+    private BigDecimal insuranceCostPercent;
+
+    public DailyInterestPolicyWithInsurance(BigDecimal yearlyInterestRatePercent, BigDecimal insuranceCostPercent) {
+        this.yearlyInterestRatePercent = yearlyInterestRatePercent;
+        this.insuranceCostPercent = insuranceCostPercent;
+    }
+
+    @Override
+    public BigDecimal getYearlyInterestRatePercent() {
+        return this.yearlyInterestRatePercent;
+    }
+
+    @Override
+    public BigDecimal calculateInterest(BigDecimal amount, LocalDate openDate, LocalDate calculationDate, LocalDate plannedTerminationDate) {
+        int depositDuration = getDuration(openDate, plannedTerminationDate.isBefore(calculationDate) ? plannedTerminationDate : calculationDate);
```

Comment:
Consider extracting "plannedTerminationDate.isBefore(calculationDate) ? plannedTerminationDate : calculationDate" to private method.
Write unit test where calculationDate is after plannedTerminationDate to check if interest is calculated properly.